src/components/RankingSexTagItem.tsx

```diff
@@ -1,28 +1,45 @@
+import StyledRankingSexTagItemBtn from '@/styles/StyledRankingSexTagItemBtn';
 import styled from '@emotion/styled';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 
 const sexTagItemlist = ['전체', '여성이', '남성이', '청소년이'];
 
 const StyledRankingSexTagItem = styled.div`
   width: 60px;
   height: 50px;
-  background-color: ${({ theme }) => theme.palette.blue100};
-  border-radius: 10px;
   display: flex;
   align-items: center;
   justify-content: center;
   margin: 10px;
-  ${({ theme }) => theme.typography.label1Regular}
-  color: ${({ theme }) => theme.sementicPalette.textDefault};
 `;
 
 const RankingSexTagItem = () => {
+  const navigate = useNavigate();
+
+  const { search } = useLocation();
+  const [selected, setSelected] = useState<string | null>('전체');
+
+  useEffect(() => {
+    const params = new URLSearchParams(search);
+    const value = params.get('SexTagSelected');
+    if (value) {
+      setSelected(value);
+    }
+  }, [search]);
+  const handleClick = (value: string) => {
+    const params = new URLSearchParams(search);
+    params.set('SexTagSelected', value);
+    navigate(`?${params.toString()}`, { replace: true });
+  };
   return (
     <>
       {sexTagItemlist.map((item: string) => {
         return (
-          <StyledRankingSexTagItem key={item} className='ranking-sex-tag-item'>
+          <StyledRankingSexTagItemBtn isSelected={selected === item} key={item} onClick={() => handleClick(item)}>
+            <StyledRankingSexTagItem className='ranking-sex-tag-item'></StyledRankingSexTagItem>
```

Comment:
self-closing tag를 사용해볼수 있겠네용!
요부분은 prettier/eslint로도 설정 가능하답니당!

```suggestion
            <StyledRankingSexTagItem className='ranking-sex-tag-item' />
```